internal/pkg/template/templates/environment/partials/cfn-execution-role.yml

```diff
@@ -14,7 +14,6 @@ CloudformationExecutionRole:
         Principal:
           Service:
           - 'cloudformation.amazonaws.com'
-          - 'lambda.amazonaws.com'
```

Comment:
@Lou1415926 can you remind us 😅

Reply:
iirc initially there was an env template version so that the `svc deploy` knows to upgrade the env template automatically. This was the `env upgrade` era.

Now, there is no `env upgrade`, and we substituted it with:
1. For an env-controller feature used in service, check if the environment that it deploys to support that feature. We do this by comparing the least env template version that supports that feature and the env's actual version.
2. Maybe for bug fixes that impact `svc deploy`
3. Encourage users to upgrade their env by running `copilot env deploy`

If this is a `chore`, not a `fix`, and it doesn't involve env controller feature, then maybe we don't need to bump the version

I also noticed that https://github.com/aws/copilot-cli/blob/bf4f19872dce711b1be277ae25407799b1d3efde/internal/pkg/template/env.go#L47 probably needed to be env template version instead of the CLI version 🤔

Reply:
Maybe we should make `envTemplate` version the same as the CLI version so that it's easier to keep track 😅 . I recall some discussion around this. Maybe we should pick this up soon.

Reply:
gotcha - we'll leave the version for this change then? i think i have the App Runner feature version sitting around in another PR https://github.com/aws/copilot-cli/pull/4150 - i'll work on getting that one merged!

Reply:
Ja I think we can leave it!